### PR TITLE
PieChart: Support row data in pie charts

### PR DESCRIPTION
**Pull request by torkelo**
It's very frustrating that the pie chart currently cannot visualize:

Screenshot from 2021-05-26 17-29-51
Screenshot from 2021-05-26 17-29-57

A few weeks ago we removed the "all values" option that is part of stat/gauge/bargauge. Because it does not work with palette based color schemes, setting manual color does not work 
either as we can only color fields, not values. 

Screenshot from 2021-05-26 17-31-35

One way would be to add a transform that pivots the data (Rows to fields). But that is a bit painful from a user / ease of use perspective. In this PR I tested a way to update seriesIndex to get unique auto colors. We could never support custom color selection with this mode. 

With this PR
Screenshot from 2021-05-26 17-38-08

The thing that would be nice to add with this an auto mode that selects Calculate or All values based on the data using some heuristic. 

Question is given that color picker will never work with the data in this form, should we skip this solution and explore a way for a panel add a transform? but how to make that easy to use? Or panel edit to suggest a transform? 

Updated so that color picker created overrides does work. They will say "not found" in panel edit but this feels ok

### CODE REVIEW

**Comment by torkelo** [comment]
Ok, added logic that makes color overrides added by the color picker work

File: packages/grafana-data/src/field/fieldDisplay.ts [hunk 1]
```diff
@@ -152,6 +152,13 @@ export const getFieldDisplayValues = (options: GetFieldDisplayValuesOptions): Fi
             }
           }
 
+          // Palette color modes use series index (field index) which does not work for when displaing rows
+          // So updating seriesIndex here makes the palette color modes work in "All values" mode
+          field.state = {
```
Comment (mckn) [inline packages/grafana-data/src/field/fieldDisplay.ts:157]: Nit: would it be worth to change this to something like `field.state = setIndexForPaletteColor(field.state)` and move the code + comment into a small help function.

**Review by mckn — APPROVED** [review]
Great work with this!

File: packages/grafana-data/src/field/fieldDisplay.ts (hunk 1 above)
Comment (torkelo) [reply packages/grafana-data/src/field/fieldDisplay.ts:157]: :+1:

**Pull request merged**